### ISSUE TITLE
Missing POSIX executable on Win10

### ISSUE THREAD

**Issue by LukasHebing** [nat-n/poethepoet#195]
I am using `poethepoet` inside a `venv` from poetry on Windows 10.
After I activated the virtual environment with `poetry shell`, I can use `poe`. However, any command with poe returns the error:

```
Error: Couldn't locate interpreter executable for ('posix',) to run shell task. Some dependencies may be missing from your system.
```

- In the virtual environment, the python path is defined (`python --version` points to the correct executable).

Is this a problem for windows systems? Can poe be used on Windows without WSL?

**Comment by nat-n** [comment]
Hi @LukasHebing,

TLDR: prefer [cmd](https://poethepoet.natn.io/tasks/task_types/cmd.html) tasks to [shell](https://poethepoet.natn.io/tasks/task_types/shell.html) tasks for portability.

Poethepoet works fine on windows with or without WSL, with one caveat explained below.

It looks like your task is failing because it's defined as a [shell](https://poethepoet.natn.io/tasks/task_types/shell.html) task, and post can't find a POSIX shell on your system. On windows this can work with WSL, or Git Bash, or anything else that puts a bash executable or similar on your path. You can also [configure the shell interpreter](https://poethepoet.natn.io/tasks/task_types/shell.html#using-a-different-shell-interpreter) to be pwsh or something else.

However shell tasks are only necessary if you want to use shell features or complex scripts (like control flow, inline variables or background tasks). If you're just running commands or python functions you can do with with [cmd](https://poethepoet.natn.io/tasks/task_types/cmd.html) tasks or [script](https://poethepoet.natn.io/tasks/task_types/script.html) tasks respectively which will run the task as a subprocess without involving a shell.

**Comment by LukasHebing** [comment]
Adding the interpreter `powershell` to the task fixed that. Thank you!
